mollie-payments-for-woocommerce/includes/mollie/wc/gateway/abstract.php

```diff
@@ -271,48 +257,56 @@ public function is_available() {
 			// Check the current (normal) order total
 			$order_total = $this->get_order_total();
 
-			// Don't check SEPA Direct Debit, as it's only available for recurring payments
-			if ( $this->id !== 'mollie_wc_gateway_directdebit' ) {
-
-				// If order total is more then zero, check min/max amounts
-				if ( $order_total > 0 ) {
-					// Validate min amount
-					if ( 0 < $this->min_amount && $this->min_amount > $order_total ) {
-						return false;
-					}
+			$filters = array (
+				'amount'       => array (
+					'currency' => get_woocommerce_currency(),
```

Comment:
Shouldn't this use the order currency? Same for the filters below.

I'm not sure if it is possible to have a webshop with multiple currencies? Probably with some sort of plugin.

I found this problem by testing with different currencies:

1. I first setup the shop with USD currency
2. I created an order with USD but did not completed the payment (order has status `Pending payment`)
3. In the settings I switch the shop to use EUR as currency
4. I opened the order from step 2 and clicked "Customer payment page"
5. The order checkout shows payment methods for currency EUR and not the payment methods for the currenct order (USD).
![screen shot 2018-06-27 at 13 53 55](https://user-images.githubusercontent.com/130915/41972345-ea8bac58-7a11-11e8-889e-431b0acdbed6.png)

Reply:
> Shouldn't this use the order currency? Same for the filters below.

That's not possible, the order isn't place at that point in time. Multi-currency plugins will hook into get_woocommerce_currency() and give us the correct one for that user.

> I'm not sure if it is possible to have a webshop with multiple currencies? Probably with some sort of plugin.

You'll need to use one of the plugins on this list:
https://github.com/mollie/WooCommerce/issues/203

> I found this problem by testing with different currencies:

Checking this.